eng/versioning.targets

```diff
@@ -50,11 +58,14 @@
     <_unsupportedOSPlatforms Include="$(UnsupportedOSPlatforms)" />
   </ItemGroup>
 
-  <Target Name="AddUnsupportedOSPlatformAttribute" BeforeTargets="GenerateAssemblyInfo" AfterTargets="PrepareForBuild" Condition="'@(_unsupportedOSPlatforms)' != '' and '$(IsTestProject)' != 'true' and '$(IncludePlatformAttributes)' != 'false'">
+  <Target Name="AddUnsupportedOSPlatformAttribute" BeforeTargets="GenerateAssemblyInfo" AfterTargets="PrepareForBuild"
+    Condition="'@(_unsupportedOSPlatforms)' != '' and '$(IsTestProject)' != 'true' and '$(IncludePlatformAttributes)' != 'false' and '$(TargetFrameworkIdentifier)' != '.NETFramework'">
     <ItemGroup>
       <AssemblyAttribute Include="System.Runtime.Versioning.UnsupportedOSPlatform">
         <_Parameter1>%(_unsupportedOSPlatforms.Identity)</_Parameter1>
       </AssemblyAttribute>
+      <SupportedPlatform Include="browser" Condition="'$(TargetFrameworkIdentifier)' == '.NETStandard' or 
+        ('$(TargetFrameworkIdentifier)' == '.NETCoreApp' and $([MSBuild]::VersionLessThan($(TargetFrameworkVersion), '5.0')))"/>
     </ItemGroup>
   </Target>
```

Comment:
I don't fully understand this change; can you describe what's happening here?

Reply:
This is where the assembly level `UnsupportedOSPlatform("browser")` attribute added, for lower builds there is a warnings: `"browser" is not known platform name` because there OS platform methods and no MSbuild SupportedPlatofrms list, so adding "browser" to SupportedPlatforms list for lower netstandard and netcoreapp buiilds

Reply:
Ah, I see; thanks. This feels out of place then. Instead of adding this `<SupportedPlatform>` within this target, could it be added up above by changing the condition around the other case when `browser` is added?

Reply:
In the above `<SupportedPlatform Include="browser">` added only for enabling browser warnings for browser targeted or multitargeted build, this actually affected the multitargeted builds of lower targets like `netstandard2.0` but there is still warnings for targeted builds like `netstandard2.0-Linux-Debug`, we wouldn't like to enable warnings for `linux` build. Second not all assemblies have  `UnsupportedOSPlatform("browser")`, i added here because this is where assembly level  `UnsupportedOSPlatform("browser")` attribute added. 

But after giving into it another look instead of adding <SupportedPlatform> i think i can restrict adding the `UnsupportedOSPlatform("browser")` for platform targeted builds, we only need that for cross platform builds

Reply:
I'm not sure I fully follow you here but I think we have cases where UnsupportedOSPlatform("browser") is used for the browser-specific build.

Reply:
Yes, and they are using cross-platform build results/dlls, right? We don't need the attribute for targeted assemblies of the source, right? For example System.IO.Ports src has `netstandard2.0-windows;netstandard2.0-Linux;netstandard2.0-OSX;netstandard2.0;net461-windows;netstandard2.0-FreeBSD` targets, before this change all or the build results had the  `UnsupportedOSPlatform("browser")` attribute and we are having warnings for targeted builds `netstandard2.0-windows;netstandard2.0-Linux;netstandard2.0-OSX;net461-windows;netstandard2.0-FreeBSD`, after this change the cross-platform target `netstandard2.0;` will still have the `UnsupportedOSPlatform("browser")` attribute but not the targeted ones. ref assembly has only `netstandard2.0;net461` and `netstandard2.0` ref will still have the attribute

Reply:
Right, but it could be needed for net6-browser. If this covers that then we should be good except Catalyst but that should be rare.

Reply:
This only applies to adding assembly-level Unsupported attributes, doesn't affect API level attributes, i believe when the entire assembly is unsupported on the browser we will not have `net6-browser` target for such assembly.  Projects having `net6-browser` target might reference those unsupported assemblies cross-platform build/dll which will have the `UnsupportedOSPlatform("browser")` and would get a warning for such reference